oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/InMemoryOAuth2AuthorizationService.java

```diff
@@ -63,6 +68,27 @@ public OAuth2Authorization findByToken(String token, @Nullable TokenType tokenTy
 				.orElse(null);
 	}
 
+	@Nullable
+	@Override
+	public OAuth2Authorization findByTokenWithHint(String token, @Nullable TokenType tokenTypeHint) {
+		Assert.hasText(token, "token cannot be empty");
+		List<TokenType> supportedTokenTypes = new ArrayList<>(
+				Arrays.asList(
+						TokenType.ACCESS_TOKEN,
+						TokenType.REFRESH_TOKEN,
+						new TokenType(OAuth2AuthorizationAttributeNames.STATE)));
+		if (tokenTypeHint != null && supportedTokenTypes.remove(tokenTypeHint)) {
```

Comment:
The `TokenType` is a hint; that means it will simply indicate with which supported token type we should start